File: 0169-majority-element/0169-majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int cand=0;
        int c=0;
        for(auto x:nums){
            if(c==0)
            cand=x;
            if(cand==x)
            c++;
            else
            c--;
        }
        return cand;
    }
};

File: 0169-majority-element/0169-majority-element.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        mp={}
        n=len(nums)
        for x in nums:
            if x in mp:
                mp[x]+=1
                if(mp[x]>n//2):
                    return x
            else:
                mp[x]=1
        else:
            return nums[0]
        
